src/eval_cache.rs

```diff
@@ -133,23 +131,15 @@ impl CacheManifest {
         }
 
         // Now discard all the paths who don't have an associated formatter
-        for (name, _) in self.matches {
-            if !new_formatters.contains_key(&name) {
-                new_paths.remove(&name);
-            }
-        }
-
-        // Replace with the new info
-        Self {
-            formatters: new_formatters,
-            matches: new_paths,
+        for name in old_formatters.keys() {
+            self.matches.remove(&name);
         }
     }
 
     /// Returns a new map with all the paths that haven't changed
     #[must_use]
     pub fn filter_matches(
-        self,
+        &self,
```

Comment:
That's all it took to get rid of a significant clone, because we're only reading from `self` and we don't need to "consume" it, we can just use a reference